### PR TITLE
Update deprecated antd <Menu>

### LINKED ISSUE #6812: Do not use deprecated 'children' property of antd's menu component
## Context

We are using the deprecated children property of antd's menu component.

## Expected Behavior

There should not be a warning in the console. We should do the migration suggested by the error message.
## Current Behavior

image

## Steps to Reproduce the bug

1. Open a tracing. The error should show up in the console.

### PR DESCRIPTION
**Pull request by hotzenklotz**
We regularly have depreacation warning in the browser dev tools that antd's `<Menu>{children}</Menu>` HOC has been replaced with an `items` prop: `<Menu items={...}>`. 

The recommend upgrade path from antd v4 to v5 is to fix all these deprecation warnings. In this pull request I fixed the `<Navbar>` and `<FolderTree>` menus.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Click on all navbar menu items. to make sure that they still work
- Consider styling changes (menu should look the same)

### Issues:
- fixes #6812
- Contributes to #6861

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Removed dev-only changes like prints and application.conf edits
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [ ] Needs datastore update after deployment

### CODE REVIEW

**Comment by hotzenklotz** [comment]
Mhm, I just realized that this CI error did not occur locally since I already upgrade to the latest antd v4.28.x locally. It seems like the `menu` prop was introduced in the meantime:

https://app.circleci.com/pipelines/github/scalableminds/webknossos/10348/workflows/6838ef9d-90ac-4362-95fa-73319c4572ae/jobs/21737

File: frontend/javascripts/navbar.tsx [hunk 1]
```diff
@@ -173,236 +156,243 @@ function UserInitials({
...
+  isAdminOrTeamManager: boolean,
+  collapse: boolean,
+  openHelpModal: MenuClickEventHandler,
+) {
+  const polledVersionString =
+    polledVersion != null && polledVersion !== version
+      ? `(Server is currently at ${polledVersion}!)`
+      : "";
+
+  const helSubMenuItems: ItemType[] = [
```
Comment (philippotto) [inline frontend/javascripts/navbar.tsx:300]: should be `helpSubMenuItems`

File: frontend/javascripts/navbar.tsx [hunk 2]
```diff
@@ -518,73 +507,90 @@ function LoggedInAvatar({
...
+      subMenuCloseDelay={subMenuCloseDelay}
+      triggerSubMenuAction="click"
+      items={[
+        {
+          key: "loggedMenu",
+          label: <UserInitials activeUser={activeUser} isMultiMember={isMultiMember} />,
+          style: {
+            padding: 0,
+          },
+          // className="sub-menu-without-padding vertical-center-flex-fix"
```
Comment (philippotto) [inline frontend/javascripts/navbar.tsx:525]: can this be removed?

**Review by philippotto — CHANGES_REQUESTED** [review]
Thank you for taking care of the deprecations :1st_place_medal: 

The code looks good mostly and testing went well, too. However, when making the browser window quite small, there is an unnecessary large margin here:

image

**Comment by hotzenklotz** [comment]
Thanks for the review. I applied your feedback.

File: frontend/javascripts/navbar.tsx [hunk 3]
```diff
@@ -173,236 +156,243 @@ function UserInitials({
...
-  title,
-  collapse,
-  icon,
-}: {
-  title: string;
-  collapse: boolean;
-  icon: any;
-}) {
-  if (collapse) {
-    return <span title={title}>{icon}</span>;
```
[the remark this answers is not in the document]
Comment (philippotto) [reply frontend/javascripts/navbar.tsx:186]: I think this is still missing? There are no tooltips in the tracing view with a small screen (see screenshot):
image

File: frontend/javascripts/navbar.tsx (hunk 2 above)
Comment (philippotto) [reply frontend/javascripts/navbar.tsx:526]: :wastebasket: probably?

File: frontend/javascripts/navbar.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (hotzenklotz) [reply frontend/javascripts/navbar.tsx:186]: I added a <ToolTip>. Should be fixed now.

**Review by philippotto — APPROVED** [review]
Great :+1:

**Pull request merged**